server/controllers/request.js

```diff
@@ -2,57 +2,39 @@ const Request = require("../models/Request");
 const asyncHandler = require("express-async-handler");
 
 exports.requestList = asyncHandler(async (req, res, next) => {
-  let requests;
-
-  requests = await Request.find({});
-
-  if (!requests) {
-    res.status(404);
-    throw new Error("No requests found");
-  }
+  const userId = req.user.id;
+  const requests = await Request.find({ userId: userId });
 
   res.status(200).json({ requests: requests });
 });
 
 exports.requestCreate = asyncHandler(async (req, res, next) => {
-  const { userId, sitterId, start, end } = req.body
-
-  Request.create({
+  const { sitterId, start, end } = req.body;
+  const userId = req.user.id;
+  
+  const request = await Request.create({
     userId,
     sitterId,
     start: new Date(start),
     end: new Date(end),
-  },(err, result)=>{
-    if (err) {
-      res.status(404).json({ error: 'Error occurred'});
-      throw new Error(`Error occurred: ${err}`);
-    }
   });
 
-  res.status(200).json({ message: 'request created'});
+  res.status(200).json({
```

Comment:
Forgot to mention this before but we can use the `201` status code here instead 👍🏼